src/rgw/rgw_rest_s3.cc

```diff
@@ -4648,6 +4649,16 @@ RGWAuthApplier::aplptr_t RGWS3V2LocalAuthEngine::authenticate() const
   }
   RGWAccessKey& k = iter->second;
 
+  if (!k.subuser.empty()) {
```

Comment:
I would say [this isn't the best place place for modifying `req_state`](https://github.com/mikulely/ceph/blob/5231f8191382740a1c9193ab8d1232aae01477ca/src/rgw/rgw_auth.h#L237). If we want to stay with the idea of extending it, then `modify_request_state` of `RGWLocalAuthApplier` (soonly `rgw::auth::LocalApplier`, I hope) would be our way to go.